dashboard-app/src/navigation/pages.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { Route } from 'react-router-dom';
+import { Dictionary, filter } from 'ramda';
+import Dashboard from '../dashboard/Dashboard';
+import ByActivity from '../dashboard/ByActivity';
+import ByTime from '../dashboard/ByTime';
+import ByVolunteer from '../dashboard/ByVolunteer';
+import Login from '../auth/pages/Login';
+import ResetPassword from '../auth/pages/ResetPassword';
+import ForgotPassword from '../auth/pages/ForgotPassword';
+import ErrorPage from '../Error';
+
+
+export type Page = {
+  url: string
+  title: string
+  component: React.ComponentClass | React.FunctionComponent
+  protected?: boolean
+};
+
+type PagesDictionary = Dictionary<Page>;
+
+
+export const PagesDict: PagesDictionary = {
+  Dashboard: {
+    url: '/',
+    component: Dashboard,
+    title: 'Dashboard',
+    protected: true,
+  },
+  Activity: {
+    url: '/activity',
+    component: ByActivity,
+    title: 'Activity',
+    protected: true,
+  },
+  Time: {
+    url: '/time',
+    component: ByTime,
+    title: 'Time',
+    protected: true,
+  },
+  Volunteer: {
+    url: '/volunteer',
+    component: ByVolunteer,
+    title: 'Volunteer',
+    protected: true,
+  },
+  Login: {
+    url: '/login',
+    component: Login,
+    title: 'Login',
+  },
+  ResetPassword: {
+    url: '/password/reset/:token',
+    component: ResetPassword,
+    title: 'Reset Password',
+  },
+  ForgotPassword: {
+    url: '/password/forgot',
+    component: ForgotPassword,
+    title: 'Forgot Password',
+  },
+  ErrorPage: {
+    url: '/error/:code',
+    component: ErrorPage,
+    title: 'Error',
+  },
+};
+
+
+const filterBy = (k: keyof Page, v: string) =>
+  filter((page) => page[k] === v, PagesDict);
+
+
+export const Pages = {
+  matchPath (pathname: string) {
+    const pages = filterBy('url', pathname);
+    const keys = Object.keys(pages);
+
+    if (keys.length !== 1) {
+      throw new Error(`One page expected, ${keys.length} returned: ${keys}. ${pathname}`);
+    }
+
+    return pages[keys[0]];
+  },
+
+  toRoute (page: Page) {
+    return <Route exact path={page.url} component={page.component} key={page.url} />;
```

Comment:
This is assumes that we will always be rendering pages identically, removing our ability to use alternative methods available with `react-router-dom`. We should be careful with putting abstraction layers on top of frameworks as it ties us into their current api and doesn't leave room for flexibility. 

I think abstraction on top of our own domain logic, but IMO the abstraction in this PR goes a little too far.

Reply:
yeah fair, this is mostly a utility method, but suppose it can just live in `App.tsx` instead